### DISCUSSION TITLE
[Nextjs 13] Migrate to app/layout.tsx from _app.tsx

### DISCUSSION THREAD

**Discussion by kitadesign** [nextauthjs/next-auth#5766] [Help]
I wanna use Nextjs13 + NextAuthJs.
How do i use SessionProvider at app/layout.tsx.

**Comment by kirtan-desai** [comment]
I made a client component `Provider.jsx`

```js
'use client'
import { SessionProvider } from "next-auth/react";

export default function Provider({ children, session }) {
    return (
        <SessionProvider session={session}>
            {children}
        </SessionProvider>
    );
}
```

and wrapped it around `{children}` in layout.jsx
```js
<Provider>{children}</Provider>
```

then I use `unstable_getServerSession()` from `next-auth/next` to get the session in `page.jsx`
```js
export default async function Home() {
    const session = await unstable_getServerSession();
    return session ? <LoggedInPage /> : <LoggedOutPage />;
}
```

**Reply by kitadesign** [reply]
Thanks for your answer.
`Provider` receives `session` in property.

How to create a session with `layout.jsx` ??

**Reply by shawnmclean** [reply]
@kitadesign have you found this out?

**Reply by nos-nart** [reply]
Found a working example of next 13 appDir, next-auth [taxonomy](https://github.com/shadcn/taxonomy/blob/main/app/layout.tsx)

**Comment by kirtan-desai** [comment]
Hey, actually one way to go about this is to not use SessionProvider at all. Call unstable_getServerSession() in server components. Is there any reason to use SessionProvider?

**Reply by kitadesign** [reply]
Hi, kirtan-desai.
I use Firebase Auth, and I wanna get IdToken on client side, and I wanna use IdToken on server side.
Do I need to use SessionProvider? ?
